mass_mailing_list_dynamic/models/mail_mass_mailing_contact.py

```diff
@@ -9,13 +9,15 @@
 class MassMailingContact(models.Model):
     _inherit = "mail.mass_mailing.contact"
 
-    @api.constrains("partner_id", "list_id", "name", "email")
+    @api.constrains("partner_id", "list_ids", "name", "email")
     def _check_no_manual_edits_on_fully_synced_lists(self):
         if self.env.context.get("syncing"):
             return
-        if any((one.list_id.dynamic and
-                one.list_id.sync_method == "full") for one in self):
-            raise ValidationError(
-                _("Cannot edit manually contacts in a fully "
-                    "synchronized list. Change its sync method or execute "
-                    "a manual sync instead."))
+        for one in self:
```

Comment:
I think code inside here would get easier with:

```suggestion
        for one in self.mapped("list_ids"):
```

Reply:
Not sure, what happens if it's not a singleton?

Reply:
`mapped` serves for multiple records as well